### DISCUSSION TITLE
'DEFAULT_EXTENSIONS' is referenced directly or indirectly in its own type annotation.

### DISCUSSION THREAD

**Discussion by babakfp** [open-circle/valibot#476] [Q&A] [answered]
Hi 👋

![image](https://github.com/fabian-hiller/valibot/assets/44144724/9a6111e4-6bf2-47e4-95f6-3b23655df6bf)

https://stackblitz.com/edit/vitejs-vite-xz8eex?file=src%2Fcounter.ts

I don't understand why this is happening. This isn't an issue with Valibot because the same thing happens with Zod too.

I found this https://github.com/fabian-hiller/valibot/issues/98 but I don't know what to do.

**Comment by fabian-hiller** [comment, marked as answer]
The problem is that you are defining a type with itself. The interdependency cannot be resolved by the TypeScript language server. Here is a simple fix:

```ts
import * as v from 'valibot';

const abc: string[] = ['.svelte.md'];

const ConfigSchema = v.optional(
  v.object({
    extensions: v.optional(
      v.array(v.string([v.regex(/^\.[a-z]+(\.[a-z]+)?$/)]), [v.minLength(1)]),
      abc
    ),
  }),
  {}
);

type Config = v.Output<typeof ConfigSchema>;
```
